modules/appeals_api/app/models/concerns/appeals_api/add_error.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module AddError
+    extend ActiveSupport::Concern
+
+    included do
+      def add_error(message, missing_fields: [], title: 'Missing fields', pointer: nil, code: nil)
+        error_options = { source: { pointer: pointer },
+                          detail: message,
+                          title: title,
+                          code: code,
+                          meta: { missing_fields: missing_fields } }
```

Comment:
Okay! This is something that I think is benefiting from some ruby behavior that works out in a sort of esoteric way that I think we should be more deliberate about than relying on parsing/language features to accomplish.

IF I'm understanding things correctly, we have both a keyword argument `missing_fields` as well as a private method `missing_fields` and it is hard to tell in which circumstance one will take the place of the other. If I supply values into the keyword argument, I would expect to get those, but if I don't I should be getting an empty array - yet somehow it's calling the private method instead. That's a bit too much guesswork for my liking, so let's be more deliberate -

* Rename the private method so it's obvious when it's being used vs the keyword argument.
e.g. (renaming the private to `missing_header_fields`)
`meta: { missing_fields: missing_fields.presence || missing_header_fields }`